cloudinit/url_helper.py

```diff
@@ -575,7 +575,7 @@ def read_url_handle_exceptions(
         time_taken = int(time.time() - start_time)
         max_wait_str = "%ss" % max_wait if max_wait else "unlimited"
         status_msg = "Calling '%s' failed [%s/%s]: %s" % (
-            url,
+            url or getattr(url_exc, "url", None),
```

Comment:
Instead of `None`, can we use something like `url`? `None` in a log message will make me smell a logic error.